File: mtproto/src/main/java/com/walhalla/mtprotolist/entity/MtprotoProxy.java
package com.walhalla.mtprotolist.entity;

import androidx.annotation.Keep;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;
import com.walhalla.mtprotolist.LessonState;

import java.io.Serializable;

@Keep
public class MtprotoProxy implements Serializable {


    public Long _id;

    @SerializedName("host")
    @Expose
    public String host = "";
    @SerializedName("port")
    @Expose
    public String port = "";
    @SerializedName("secret")
    @Expose
    public String secret = "";
    @SerializedName("code")
    @Expose
    public String code = "";
    @SerializedName("enabled")
    @Expose
    public Boolean enabled;

    public void setCity(String city) {
        this.city = city;
    }

    public void setRegionName(String regionName) {
        this.regionName = regionName;
    }

    public void setLat(double lat) {
        this.lat = lat;
    }

    public void setLon(double lon) {
        this.lon = lon;
    }

    @SerializedName("city")
    @Expose
    public String city;
    @SerializedName("regionName")
    @Expose
    public String regionName;
    @SerializedName("lat")
    @Expose
    public double lat;
    @SerializedName("lon")
    @Expose
    public double lon;

    @SerializedName("country")
    @Expose
    public String country;
    @SerializedName("region")
    @Expose
    public String region;
    @SerializedName("zip")
    @Expose
    public String zip;
    @SerializedName("timezone")
    @Expose
    public String timezone;
    @SerializedName("isp")
    @Expose
    public String isp;

    @SerializedName("org")
    @Expose
    public String org;

    @SerializedName("as")
    @Expose
    public String as;


//    @SerializedName("lock")
//    @Expose
//    private int lock;


    public MtprotoProxy(String host, String port, String secret, String code, Boolean enabled) {
        this.host = host;
        this.port = port;
        this.secret = secret;
        this.code = code;
        this.enabled = enabled;
    }

    public MtprotoProxy() {
    } //FB


//    public int getLock() {
//        return lock;
//    }
//
//    public void setLock(int lock) {
//        this.lock = lock;
//    }

}

File: loader/src/main/java/com/walhalla/mtprotolist/entity/MtprotoProxy.kt
package com.walhalla.mtprotolist.entity

import androidx.annotation.Keep
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import java.io.Serializable

@Keep
class MtprotoProxy : Serializable {
    var _id: Long? = null

    @SerializedName("host")
    @Expose
    var host: String? = ""

    @SerializedName("port")
    @Expose
    var port: String? = ""

    @SerializedName("secret")
    @Expose
    var secret: String? = ""

    @SerializedName("code")
    @Expose
    var code: String? = ""

    @SerializedName("enabled")
    @Expose
    var enabled: Boolean? = null


    @SerializedName("update_at")
    @Expose
    var update_at: Long = 0


    @SerializedName("city")
    @Expose
    var city: String? = null

    @SerializedName("regionName")
    @Expose
    var regionName: String? = null

    @SerializedName("lat")
    @Expose
    var lat: Double = 0.0

    @SerializedName("lon")
    @Expose
    var lon: Double = 0.0

    @SerializedName("country")
    @Expose
    var country: String? = null

    @SerializedName("region")
    @Expose
    var region: String? = null

    @SerializedName("zip")
    @Expose
    var zip: String? = null

    @SerializedName("timezone")
    @Expose
    var timezone: String? = null

    @SerializedName("isp")
    @Expose
    var isp: String? = null

    @SerializedName("org")
    @Expose
    var org: String? = null

    @SerializedName("as")
    @Expose
    var `as`: String? = null


    @SerializedName("lock")
    @Expose
    var lock: Int = 0


    constructor(host: String?, port: String?, secret: String?, code: String?, enabled: Boolean?) {
        this.host = host
        this.port = port
        this.secret = secret
        this.code = code
        this.enabled = enabled
    }

    constructor() //FB
}
